packages/protocol/contracts-0.8/common/GasPriceMinimum.sol

```diff
@@ -165,6 +165,7 @@ contract GasPriceMinimum is
       uint256 rateNumerator;
       uint256 rateDenominator;
       (rateNumerator, rateDenominator) = sortedOracles.medianRate(tokenAddress);
+      require(rateDenominator > 0, "Token has no rate");
```

Comment:
wouldn't it be cleaner for `medianRate` to do this require? If not we make every caller need to make this check

Reply:
actually... checking sorted oracles. var names looks odd

it should be:
```
rate, numOfRates = 
```

Reply:
I moved it to `medianRate` and also renamed variable to `medianRate` and `denominator`

Reply:
> wouldn't it be cleaner for `medianRate` to do this require?

I think it makes sense for the oracle to have a non-reverting version of this function so consumers can decide what to do in the case of a missing rate.

Reply:
After checking blockchain client code:

Our options are:

- Revert when no medianRate exists - problem is that [exchange rate of fee currency is consider 1:1](https://github.com/celo-org/celo-blockchain/blob/6388f0ec88fb7a2a82ee41b0e2c9cb7e2cff87e2/contracts/currency/currency.go#L228C2-L234C3) to CELO which will cause user to pay less/more than what they agreed to
- Keep it as it is (== returning 0) - problem is that if we whitelist new fee currency, but forget to add oracle feed, then we[ will panic the client ](https://github.com/celo-org/celo-blockchain/blob/6388f0ec88fb7a2a82ee41b0e2c9cb7e2cff87e2/contracts/currency/currency.go#L132C2-L132C85)since we are dividing by 0 and it has potential to shutdown whole Celo network
- Return uint256.max - which would make paying by this feeCurrency extremely expensive - we need to check whether there are no potential overflows

We agreed with @karlb to go with 3. for now